EcoSoapBank/Pickups/NewPickupViewController.swift

```diff
@@ -0,0 +1,263 @@
+//
+//  NewPickupViewController.swift
+//  EcoSoapBank
+//
+//  Created by Jon Bash on 2020-08-17.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import UIKit
+import SwiftUI
+import Combine
+
+
+class NewPickupViewController: UIViewController {
+    typealias Snapshot = NSDiffableDataSourceSnapshot<Int, NewCartonViewModel>
+
+    private var viewModel: NewPickupViewModel
+    private var cancellables: Set<AnyCancellable> = []
+
+    private lazy var tableViewHeight: NSLayoutConstraint =
+        tableView.heightAnchor.constraint(equalToConstant: 0)
+
+    // MARK: - Views
+
+    private lazy var cartonsLabel = configureSectionLabel(titled: "Cartons")
+    private lazy var dateLabel = configureSectionLabel(titled: "Pickup Date")
+    private lazy var notesLabel = configureSectionLabel(titled: "Notes")
+
+    private lazy var tableView = configure(UITableView()) {
+        $0.register(PickupCartonCell.self,
+                    forCellReuseIdentifier: PickupCartonCell.reuseIdentifier)
+        $0.delegate = self
+    }
+    private lazy var dataSource = DataSource(
+        tableView: tableView,
+        cellProvider: cell(for:at:with:))
+    private lazy var addCartonButton = configure(UIButton()) {
+        $0.setImage(UIImage.addBoxSymbol.withTintColor(.white), for: .normal) // TODO: replace with simple + button
+        $0.tintColor = .white
+        $0.imageView?.contentMode = .scaleAspectFit
+        $0.layer.cornerRadius = 5
+        $0.backgroundColor = .esbGreen
+        $0.addTarget(self, action: #selector(addCarton), for: .touchUpInside)
+        $0.contentEdgeInsets = configure(UIEdgeInsets(), with: { ei in
+            ei.top = 4
```

Comment:
No, I think Lambda School (or SwiftLint's defaults) might have disabled that rule of > 2 characters, perhaps since Apple has started using `id` and some other short variable names in their APIs. And I think since it's declared right there and only used in the closure, and the type if it is also right there, it makes it relatively obvious IMO what it's referring to. That's my excuse at least. 😅